prov/efa/src/rdm/rxr_ep.c

```diff
@@ -1098,6 +1101,75 @@ static int rxr_ep_set_cuda_api_permitted(struct rxr_ep *ep, bool cuda_api_permit
 	return 0;
 }
 
+/**
+ * @brief set use_device_rdma flag in rxr_ep.
+ *
+ * If the environment variable FI_EFA_USE_DEVICE_RDMA is set, this function will
+ * return an error if the value of use_device_rdma is in conflict with the
+ * environment setting.
+ *
+ * @param[in,out]	ep			endpoint
+ * @param[in]		use_device_rdma		when true, use device RDMA capabilities.
+ * @return		0 on success
+ *
+ * @related rxr_ep
+ */
+static int rxr_ep_set_use_device_rdma(struct rxr_ep *ep, bool use_device_rdma)
+{
+	bool env_value, env_set;
+
+	uint32_t api_version =
+		 rxr_ep_domain(ep)->util_domain.fabric->fabric_fid.api_version;
+
+	env_set = rxr_env_is_set_use_device_rdma();
+	if (env_set) {
+		env_value = rxr_env_get_use_device_rdma();
+	}
+
+	if FI_VERSION_LT(api_version, FI_VERSION(1, 18)) {
+		/* let the application developer know something is wrong */
+		EFA_WARN( FI_LOG_EP_CTRL,
+			"Applications using libfabric API version <1.18 are not "
+			"allowed to call fi_setopt with FI_OPT_EFA_USE_DEVICE_RDMA.  "
+			"Please select a newer libfabric API version in "
+			"fi_getinfo during startup to use this option.\n");
+		return -FI_ENOPROTOOPT;
+	}
+
+	if (env_set && use_device_rdma && !env_value) {
```

Comment:
can we just check `use_device_rdma != env_value`?

Reply:
No.  For instance, the application can set use_device_rdma off, but the default environment variable could have been on, but not explicitly set.  We want to ensure we only enter the `if` when the user has explicitly set the variable to be in conflict with the provided argument.